src/index.ts

```diff
@@ -15,78 +18,108 @@ export function isType<P>(
   return action.type === actionCreator.type;
 }
 
+export function isError<P, E extends Error>(action: Action<P | E>): action is Action<E> {
+  return action.error
+}
+
 export interface ActionCreator<P> {
   type: string;
-  (payload?: P, meta?: Object): Action<P>;
+  (payload: P, meta?: Object): Action<P>;
+}
+
+export interface ActionCreatorWithoutPayload {
+  type: string;
+  (payload?: undefined, meta?: Object): Action<undefined>;
+}
+
+
+export interface DonePayload<P, R> {
+  params: P;
+  result: R;
+}
+
+export interface FailedPayload<P, E> {
+  params: P;
+  error: E;
 }
 
-export interface AsyncActionCreators<P, R> {
+export interface AsyncActionCreators<P, R, E> {
   type: string;
   started: ActionCreator<P>;
-  done: ActionCreator<{
-    params: P;
-    result: R;
-  }>;
-  failed: ActionCreator<{
-    params: P;
-    error: any;
-  }>;
+  done: ActionCreator<DonePayload<P, R>>;
+  failed: ActionCreator<FailedPayload<P, E>>;
 }
 
 export interface ActionCreatorFactory {
-  <P>(type: string, commonMeta?: Object, error?: boolean): ActionCreator<P>;
-
-  async<P, S>(type: string, commonMeta?: Object): AsyncActionCreators<P, S>;
+  (type: string): ActionCreatorWithoutPayload;
+  <P>(type: string, commonMeta?: Object): ActionCreator<P>;
+  <P>(type: string, commonMeta: Object | undefined, isError: boolean): ActionCreator<P>;
+  <P, E>(type: string, commonMeta: Object | undefined, isError: (payload: P | E) => boolean): ActionCreator<P | E>;
+  <P, E extends Error>(type: string, commonMeta?: Object): ActionCreator<P | E>;
+  
+
+  async<P, S, E>(type: string, commonMeta?: Object): AsyncActionCreators<P, S, E>;
 }
 
+function wrapIsError(default_: ((payload: any) => boolean), isError?: ((payload: any) => boolean) | boolean) {
+  if (isError === undefined) {
+    return default_
+  }
+  if (typeof isError === "boolean") {
+    return () => isError
+  }
+  return isError
+}
 
-export default function actionCreatorFactory(prefix?: string):
+export function actionCreatorFactory(prefix?: string, defaultIsError?: ((payload: any) => boolean) | boolean):
```

Comment:
Only if you want to completely opt out of `payload instanceof Error`. It's an advanced feature so it shouldn't wrap booleans (providing a lambda is easier to understand), if it's kept at all.
There's always the trade of between flexibility and simplicity.